lib/graphql/subscriptions/serialize.rb

```diff
@@ -33,8 +33,8 @@ def dump_recursive(obj)
         case
         when obj.is_a?(Array)
           obj.map { |i| dump_recursive(i) }.join(':')
-        when obj.is_a?(Hash)
-          obj.map { |k, v| "#{dump_recursive(k)}:#{dump_recursive(v)}" }.join(":")
+        when obj.respond_to?(:to_h)
```

Comment:
I agree that this is probably not the safest way to fix this and it is probably worth looking for a different/better solution.

I wish moving the branch down would work, but unfortunately it doesn't because `GraphQL::Schema::InputObject` responds to `to_param`:

```
[7] pry(GraphQL::Subscriptions::Serialize)> obj[2]['stream'].respond_to?(:to_param)
=> true
[8] pry(GraphQL::Subscriptions::Serialize)> obj[2]['stream'].to_param
=> "#<ClassBasedInMemoryBackend::StreamInput:0x00007fe4a7c0e390>"
```

Which is what was causing the bug to begin with. It looks like maybe rails adds `to_param` to `Object`, so everything would respond to it (in `rails c`, `Module.new.respond_to?(:to_param) => true` and `Class.new.respond_to?(:to_param) => true`, but in `irb`, both are `false`). I suppose another option here might be to override `to_param` on `InputObject` if that would be safe/make sense.

Adding a new branch works for me as well - the only concern I would have about that is that it would make it more work for someone to customize how and `InputObject` gets serialized if they needed to for some reason. That may not be worth worrying about though.

What do you think?

Reply:
> more work to customize 

Can you clarify that a bit, how does it require more work?

Reply:
So with the extra branch we would say `when GraphQL::Schema::InputObject; dump_recursive(obj.to_h)`, which means that any `GraphQL::Schema::InputObject` gets dumped using `to_h`. What if you wanted to change how one specific `GraphQL::Schema::InputObject` got serialized and make it _not_ use `to_h`? You would have to override `GraphQL::Subscriptions::Serialize#dump_recursive` in order to tell it to do something different for this one specific `InputObject`.

It wouldn't be _that_ much extra work, it's just a bit more work than overriding a single method like `to_param` on the `InputObject` that you wanted to customize. Which is why I'm not sure if it's really worth worrying about or not.

Reply:
Oh, that's a fair point, although I think the usage of `InputObject#to_param` was coincidental, isn't that something that Rails adds to every object :P 

If you think it's important to support that case, we could add some _other_ method like `InputObject#to_subscription_scope`, whose default implementation is `#to_h`, then people could override it. 

But I never _expected_ anyone to customize that behavior, and I've never _heard_ of anyone doing it, so in my opinion, it would be appropriate to leave it until later, too.

Reply:
I have no idea if anyone would need to do that haha, if you've never heard of people needing to do it then it's probably safe to assume no one has needed to. I'll go with the extra branch approach!